GsaGH/Components/0_Model/GetGeometry.cs

```diff
@@ -1,92 +1,231 @@
+              .Count;
+            j++)
+            writer.SetString("dropdowncontents" + i + j, dropDownItems[i][j]);
+        }
+
+        dropdown = true;
+      }
+
+      writer.SetBoolean("dropdown", dropdown);
+
+      bool spacer = false;
+      if (spacerDescriptions != null) {
+        writer.SetInt32("spacerCount", spacerDescriptions.Count);
+        for (int i = 0; i < spacerDescriptions.Count; i++)
+          writer.SetString("spacercontents" + i, spacerDescriptions[i]);
+        spacer = true;
+      }
+
+      writer.SetBoolean("spacer", spacer);
+
+      bool select = false;
+      if (selectedItems != null) {
+        writer.SetInt32("selectionCount", selectedItems.Count);
+        for (int i = 0; i < selectedItems.Count; i++)
+          writer.SetString("selectioncontents" + i, selectedItems[i]);
+        select = true;
+      }
+
+      writer.SetBoolean("select", select);
+
+      return writer;
+    }
+
+    internal static void ReadDropDownComponents(
+      ref GH_IReader reader,
+      ref List<List<string>> dropDownItems,
+      ref List<string> selectedItems,
+      ref List<string> spacerDescriptions) {
+      if (reader.GetBoolean("dropdown")) {
+        int dropdownCount = reader.GetInt32("dropdownCount");
+        dropDownItems = new List<List<string>>();
+        for (int i = 0; i < dropdownCount; i++) {
+          int dropdowncontentsCount = reader.GetInt32("dropdowncontentsCount" + i);
+          var tempcontent = new List<string>();
+          for (int j = 0; j < dropdowncontentsCount; j++)
+            tempcontent.Add(reader.GetString("dropdowncontents" + i + j));
+          dropDownItems.Add(tempcontent);
+        }
+      }
+      else
+        throw new Exception("Component doesnt have 'dropdown' content stored");
+
+      if (reader.GetBoolean("spacer")) {
+        int dropdownspacerCount = reader.GetInt32("spacerCount");
+        spacerDescriptions = new List<string>();
+        for (int i = 0; i < dropdownspacerCount; i++)
+          spacerDescriptions.Add(reader.GetString("spacercontents" + i));
+      }
+
+      if (!reader.GetBoolean("select"))
+        return;
+
+      int selectionsCount = reader.GetInt32("selectionCount");
+      selectedItems = new List<string>();
+      for (int i = 0; i < selectionsCount; i++)
+        selectedItems.Add(reader.GetString("selectioncontents" + i));
+    }
+
     #region Name and Ribbon Layout
+
     public override Guid ComponentGuid => new Guid("6c4cb686-a6d1-4a79-b01b-fadc5d6da520");
-    public GetGeometry()
-      : base("Get Model Geometry", "GetGeo", "Get nodes, elements and members from GSA model",
-            CategoryName.Name(),
-            SubCategoryName.Cat0())
-    {
-    }
+
+    public GetGeometry() : base("Get Model Geometry",
+      "GetGeo",
+      "Get nodes, elements and members from GSA model",
+      CategoryName.Name(),
+      SubCategoryName.Cat0()) { }
+
     public override GH_Exposure Exposure => GH_Exposure.secondary;
```

Comment:
Properties should come before the constructor no?

Reply:
yup should be, but for this task I'm doing only renaming